pkg/collectsub/server/server.go

```diff
@@ -33,7 +33,9 @@ type server struct {
 	pb.UnimplementedColectSubscriberServiceServer
 
 	// Db points to the backend DB, public for mocking testing purposes.
-	Db   db.CollectSubscriberDb
+	Db db.CollectSubscriberDb
+	// Ctx is the context.Context the server is initialized with
+	Ctx  context.Context
```

Comment:
there are multiple ways to do this but probably the easiest is to store the logger in the server struct:
```
type server struct {
	pb.UnimplementedColectSubscriberServiceServer

	// Db points to the backend DB, public for mocking testing purposes.
	Db   db.CollectSubscriberDb

	logger *zap.SugaredLogger
	port int
}
``` 

This will allow access to `AddCollectEntries` and `GetCollectEntries` to use the logger from the the server struct `logger := s.logger`. Adding another context will result in confusion about which context is being used.

Reply:
i wonder if there's some construct to help set up a connection context for the grpc server

Reply:
@pxp928 Sounds good. I'll make the change.

Reply:
Based on my quick search yesterday I did not see this article: https://medium.com/hootsuite-engineering/golang-grpc-distributed-context-propagation-5809e2d66db4 but this is the better way to do it as suggested by @lumjjb.